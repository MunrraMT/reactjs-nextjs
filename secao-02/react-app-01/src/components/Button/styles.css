.btn-more-posts {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0.25rem;
  height: 3rem;
  font-size: 1.25rem;
  text-transform: uppercase;
  cursor: pointer;
}

.btn-more-posts:disabled {
  background-color: #eee;
  cursor: not-allowed;
}
