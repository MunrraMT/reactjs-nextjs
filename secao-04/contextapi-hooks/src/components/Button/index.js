const Button = ({ children, onBtnClick, disabled }) => (
  <button disabled={disabled} onClick={onBtnClick} type="button">
    {children}
  </button>
);

export default Button;
