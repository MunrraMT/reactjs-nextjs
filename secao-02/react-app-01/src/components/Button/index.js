import './styles.css';

import { Component } from 'react';

class Button extends Component {
  render() {
    const { children, handleClick, disabled } = this.props;

    return (
      <button
        disabled={disabled}
        className='btn-more-posts'
        type='button'
        onClick={handleClick}
      >
        {children}
      </button>
    );
  }
}

export default Button;
