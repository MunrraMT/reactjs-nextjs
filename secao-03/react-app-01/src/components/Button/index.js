import { func, string, bool } from 'prop-types';
import './styles.css';

const Button = ({ handleClick, disabled, text }) => (
  <button
    className='btn-more-posts'
    type='button'
    onClick={handleClick}
    disabled={disabled}
  >
    {text}
  </button>
);

Button.defaultProps = {
  disabled: false,
};

Button.propTypes = {
  handleClick: func.isRequired,
  disabled: bool,
  text: string.isRequired,
};

export default Button;
