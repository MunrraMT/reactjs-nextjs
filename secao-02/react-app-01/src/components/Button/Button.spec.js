import { fireEvent, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Button from '.';

describe('Deve testar <Button />', () => {
  const fn = jest.fn();
  const text = 'load more';

  test('Deve renderizar sem erros', () => {
    render(<Button text={text} handleClick={fn} />);
  });

  test('Deve ter uma classe - "btn-more-posts"', () => {
    const { getByRole } = render(<Button text={text} handleClick={fn} />);
    const button = getByRole('button');

    expect.assertions(1);
    expect(button).toHaveAttribute('class', 'btn-more-posts');
  });

  test('Deve conter um texto - "Load more"', () => {
    const { getByRole } = render(<Button text={text} handleClick={fn} />);
    const button = getByRole('button', { name: /load more/i });

    expect.assertions(1);
    expect(button).toBeInTheDocument();
  });

  test('Deve chamar uma função quando clicado - fireEvent', () => {
    const { getByRole } = render(<Button text={text} handleClick={fn} />);

    fireEvent.click(getByRole('button'));

    expect(fn).toHaveBeenCalledTimes(1);
  });

  test('Deve chamar uma função quando clicado - userEvent', () => {
    const { getByRole } = render(<Button text={text} handleClick={fn} />);

    userEvent.click(getByRole('button'));

    expect(fn).toHaveBeenCalledTimes(1);
  });

  test('Deve estar disabled quando disabled for passado como true', () => {
    const { getByRole } = render(
      <Button disabled text={text} handleClick={fn} />,
    );

    expect(getByRole('button')).toBeDisabled();
  });

  test('Deve estar enabled quando disabled for passado como false', () => {
    const { getByRole } = render(
      <Button disabled={false} text={text} handleClick={fn} />,
    );

    expect(getByRole('button')).toBeEnabled();
  });

  test('Deve fazer um Snapshot', () => {
    const { container } = render(
      <Button disabled={false} text={text} handleClick={fn} />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
