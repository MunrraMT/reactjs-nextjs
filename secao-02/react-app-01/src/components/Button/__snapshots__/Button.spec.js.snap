// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve testar <Button /> Deve fazer um Snapshot 1`] = `
<button
  class="btn-more-posts"
  type="button"
>
  load more
</button>
`;
